### ISSUE TITLE
Update vignettes and readme for 1.0

### ISSUE THREAD

**Issue by jgabry** [stan-dev/cmdstanr#590]
One of the last things we should do before the 1.0 release is to update the main vignette, which currently refers to CmdStanR being in beta. 

Also the readme needs to be updated

**Comment by jgabry** [comment]
Also, remember that the opencl vignette doesn't run unless the `CMDSTANR_OPENCL_TESTS` environment variable is set. And when publishing the OPENCL vignette to the website with v1.0 (and eventually to CRAN) we need to decide if we want it to be run in linux or if it's ok if I generate it on mac.

**Comment by rok-cesnovar** [comment]
Yeah, I will run it with my GPU whenever we will do that. Not a problem.

**Comment by jgabry** [comment]
The only issue is that CRAN allows each package to have only a single maintainer at a time and that person has to submit the releases. So I can't be the maintainer but have you generate one of the vignettes (as far as I know, but maybe there's a workaround). So I think that gives us a few alternatives: 

(a) I'm the CRAN maintainer and the vignettes are all generated on my mac
(b) You're the CRAN maintainer and the vignettes are all generated on whatever linux machine you're using 
(c) We don't include the vignettes with the CRAN package, just links to the ones on the CmdStanR website. This allows either of us to be CRAN maintainer and either of us to upload any of the vignettes to the CmdStanR website. 

I don't really care who the maintainer is (I'm happy to do it or not do it), so I guess if we want to make sure the opencl vignette is generated by you then we should do option (b) or (c). I'm ok with either of those options.

**Comment by rok-cesnovar** [comment]
We could also just not include the OpenCL vignette right?

**Comment by jgabry** [comment]
Yeah that's true. We could have the OpenCL vignette only be on the website and not on CRAN. That may be the best option.

**Comment by jgabry** [comment]
I will add this to #595. I think we can put the OpenCL vignette in a subdirectory and have CRAN ignore it but pkgdown put it online.

**Comment by danielinteractive** [comment]
@jgabry Hi Jonah, quick question from the sidelines - do you plan to put `cmdstanr` on CRAN soon?

**Comment by jgabry** [comment]
> do you plan to put `cmdstanr` on CRAN soon?

Yes! 

There are only a few more issues to address before we do a v1.0 release: 

https://github.com/stan-dev/cmdstanr/milestone/2

And really only two of those issues involve much work (#582 and #534).

**Comment by danielinteractive** [comment]
Hi @jgabry , so from our side we would be very interested to see `cmdstanr` on CRAN to enable downstream packages to be on CRAN too. 
What do you think - where could we help with tackling `cmdstanr` issues? I would be happy also to have a quick video call with you / the team to get started. Thanks!

**Comment by danielinteractive** [comment]
@jgabry @rok-cesnovar any thoughts/feedback?

**Comment by rok-cesnovar** [comment]
Hey, sorry, meant to respond a few days ago, but it slipped my mind. Thanks for reaching out and offering the help!! Much appreciated.

As far as getting cmdstanr on CRAN, we have a 1.0.0 milestone (https://github.com/stan-dev/cmdstanr/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22v1.0.0+-+release%22). What we are mostly lacking at this point is more documentation, vignettes and examples. Outside of that, its just a few minor features that we could also discuss bumping to future releases. I mean to do a push on some of the smaller bugfixes/features, but have been pretty busy in the last weeks.

Happy to schedule a call to discuss how we could push this over the line.

**Comment by dmuck** [comment]
@jgabry @rok-cesnovar I'd be willing to help update the documentation, vignettes, and examples, particularly for working with draws. 

It seems like there are a few related issues currently open that concern postprocessing draws (e.g., #632 , #582 , #641 ).

I'd be interested to hear more about how I could help given where things stand.

**Comment by rok-cesnovar** [comment]
I think we primarily need a few more vignettes with basic examples. So just run sampling on some model and then:
- work with samples in different posterior package formats
- use draws_rvars and the rstan-extract style with draws_of - https://github.com/stan-dev/cmdstanr/issues/632#issuecomment-1047717297
- use the default $summary() function and show how to run non-default summaries with it (similar to what is shown in  [custom summary ](https://mc-stan.org/posterior/reference/draws_summary.html#examples))

I don't think it needs to be anything fancy.

**Comment by jgabry** [comment]
Thanks for the help offers. I agree with @rok-cesnovar that's it's mostly doc improvements / vignettes that we need. Any help with those would be much appreciated! (I've been overwhelmed with other work recently so I haven't had much of a chance to do those things myself yet.)

**Comment by jgabry** [comment]
A few other things that could use more doc / vignette: 

* How to use new expose_functions()` method introduced in #702
* How to use the methods like `log_prob()` introduced in #701

**Comment by gravesti** [comment]
I've taken a stab at starting a new vignette with a few more examples for `$summary` in #791.
Did you already start something @dmuck ?

**Comment by jgabry** [comment]
> A few other things that could use more doc / vignette:
> 
> * How to use new method introduced in [Add optional method for exposing stan functions to R #702](https://github.com/stan-dev/cmdstanr/pull/702)


For this, I think right now there's no doc for how to use functions that are compiled (the `compile` method argument `compile_standalone` is documented, but not how to access the functions once they are compiled (i.e., via `mod$functions$foo()`). 

> * How to use the methods introduced in [Add optional methods for log_prob, grad_log_prob, hessian, un/constrain pars #701](https://github.com/stan-dev/cmdstanr/pull/701)

For this, there's an example of using `log_prob` at `?"fit-method-log_prob"` but the example just passes in a vector of numbers for `upars`. But just typing in the `upars` manually is pretty much impossible to do for a real example and I think we need to demonstrate how the user can use these methods together, i.e. `log_prob` in conjunction with `unconstrain_pars` (and create the structured list for `unconstrain_pars` conveniently). @andrjohns Do you already have some examples of this?

**Comment by dmuck** [comment]
> I've taken a stab at starting a new vignette with a few more examples for `$summary` in #791. Did you already start something @dmuck ?

Not yet, @gravesti. I should be able to start something next week.

**Comment by jgabry** [comment]
> I've taken a stab at starting a new vignette with a few more examples for `$summary` in #791.
> Did you already start something @dmuck ?

Thanks for starting this. Are you and @dmuck planning on working together on this or on two separate things?

**Comment by dmuck** [comment]
@jgabry We hadn't discussed working together. I was planning on putting together something that demonstrated working with draws in various formats, including draws_rvars. 

However, I'd be willing to collaborate if others have already started something.

**Comment by danielinteractive** [comment]
Hi guys, we are just talking again about `cmdstanr` getting to CRAN - anything we can help you out with this? (Background is that we have a package using `cmdstanr`, so we cannot push that to CRAN before `cmdstanr` is on CRAN :-))

**Comment by mitzimorris** [comment]
question:  where is the R community on R-universe?

**Comment by danielinteractive** [comment]
Yeah I guess for our pharma industry environment things are moving slower than elsewhere, so we would still be very motivated to get our package (and therefore `cmdstanr`) to CRAN :-) And as mentioned above, we are happy to help to work on issues etc.
